File: Module#02/model/src/main/java/com/epam/esm/domain/Certificate.java
package com.epam.esm.domain;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.Set;

@Data
@Builder
public class Certificate implements BaseEntity {

    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
    private Instant createDate;
    private Instant lastUpdateDate;
    private Integer duration;
    private Set<Tag> tags;

    @Override
    public String toString() {
        return String.format(
                "{id=%d, name='%s', description='%s', price=%s, duration=%d, tags=%s}",
                id, name, description, price, duration, tags);
    }
}

File: angular/src/app/components/model/Certificate.ts
import {Tag} from "./Tag";

export interface Certificate {
    id: string;
    name: string;
    description: string;
    shortDescription: string;
    company: string;
    price: number;
    duration: number;
    createDate: Date;
    lastUpdate: Date;
    path: string;
    tags: Set<Tag>;
}
